### PR TITLE
Feature/smartos path in method script

### PR DESCRIPTION
**Pull request by bixu**
The method script for chef-client on SmartOS needs a slightly different PATH than on Solaris or, for example, OmniOS.

### CODE REVIEW

**Comment by chef-supermarket** [comment]
Hi. I am an automated pull request bot named Curry. There are commits in this pull request whose authors are not yet authorized to contribute to Chef Software, Inc. projects or are using a non-GitHub verified email address. To become authorized to contribute, you will need to sign the Contributor License Agreement (CLA) as an individual or on behalf of your company. [You can read more on Chef's blog.](https://www.chef.io/blog/2014/06/23/changes-to-the-contributor-license-agreement-process)

## Non-GitHub Verified Committers

There are 1 commit author(s) whose commits are authored by a non-GitHub verified email address. Chef will have to manually verify that they are authorized to contribute.

[Please sign the CLA here.](https://supermarket.chef.io/become-a-contributor)

**Comment by bixu** [comment]
I have signed the CLA, so this should be safe to merge.

File: metadata.rb [hunk 1]
```diff
@@ -4,7 +4,7 @@
 license           'Apache 2.0'
 description       'Manages client.rb configuration and chef-client service'
 long_description  IO.read(File.join(File.dirname(__FILE__), 'README.md'))
-version           '4.2.4'
+version           '4.2.5'
```
Comment (jtimberman) [inline metadata.rb]: Please don't increment versions in PRs. We will change the version for releases.

File: attributes/default.rb [hunk 2]
```diff
@@ -153,6 +154,7 @@
   default['chef_client']['method_dir'] = '/opt/local/lib/svc/method'
   default['chef_client']['bin_dir'] = '/opt/local/bin'
   default['chef_client']['locale'] = 'en_US.UTF-8'
+  default['chef_client']['bin_path'] = '/usr/local/sbin:/usr/local/bin:/opt/local/sbin:/opt/local/bin:/usr/sbin:/usr/bin:/sbin'
```
Comment (jtimberman) [inline attributes/default.rb]: Let's call this new attribute, `['env_path']` to be more clear about the intent.

**Comment by bixu** [comment]
@jtimberman, made changes inline with your requests

**Comment by jtimberman** [comment]
LGTM, thanks.

**Comment by juliandunn** [comment]
:+1:

**Pull request merged**